charts/vela-core/templates/defwithtemplate/daemon.yaml

```diff
@@ -467,6 +472,9 @@ spec:
         		ip: string
         		hostnames: [...string]
         	}]
+
+        	// +usage=If addDeployVersionAnnotation is true, the deploy version annotation will be added to the underlying pods
+        	addDeployVersionAnnotation: *false | bool
```

Comment:
addPublishVersion

Reply:
sideaffect: addPublishVersion will affect all components even the component has no change.

Reply:
1. addRevisionLabel is `ignore` now, is it really deprecated now?

- vela rollout depends on it when we need traffic rolling, it don't affect vela rollout with only replica rollout.
- addRevisionLabel will automatically inject revision on component workload change.
- addRevisionLabel only restart when component workload changed while traits changes don't affect the revision.

2. the usage of addPublishVersion

- publishVersion is currently the application revision specified by user, it works through velaux to application with whole lifecycle.
- publishVersion can help restart pods when new version published, adding this lable can implement this feature. Now we don't restart pods on new release, we can't tell the pod version.
- sideaffect: no automation, need to specify unique publishVersion manually on every release.


3. Users can have work around now

 - Add labels manually.

Reply:
I would suggest making reuse of appRevision. Although PublishVersion is clear and manually controllable, there are times that user wants to make automatic changes. I wonder if appRevision could handle both the automatic case and the manual case.